include/mbedtls/pk.h

```diff
@@ -222,6 +222,28 @@ typedef struct mbedtls_pk_rsassa_pss_options {
 #define MBEDTLS_PK_USE_PSA_EC_DATA
 #endif /* MBEDTLS_USE_PSA_CRYPTO && !MBEDTLS_ECP_C */
 
+/* Internal helper to define which fields in the pk_context structure below
+ * should be used for EC keys: legacy ecp_keypair or the raw (PSA friendly)
+ * format. It should be noticed that this only affect how data is stored, not
+ * which functions are used for various operations. The overall picture looks
+ * like this:
+ * - if USE_PSA is not defined and ECP_C is then use ecp_keypair data structure
+ *   and legacy functions
+ * - if USE_PSA is defined and
+ *     - if ECP_C then use ecp_keypair structure, convert data to a PSA friendly
+ *       format and use PSA functions
+ *     - if !ECP_C then use new raw data and PSA functions directly.
+ *
+ * The main reason for the "intermediate" (USE_PSA + ECP_C) above is that as long
+ * as ECP_C is defined mbedtls_pk_ec() gives the user a read/write access to the
+ * ecp_keypair structure inside the pk_context so he/she can modify it using
+ * ECP functions which are not under PK module's control.
+ */
+#if defined(MBEDTLS_USE_PSA_CRYPTO) && defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY) && \
+    !defined(MBEDTLS_ECP_C)
+#define MBEDTLS_PK_USE_PSA_EC_DATA
+#endif /* MBEDTLS_USE_PSA_CRYPTO && !MBEDTLS_ECP_C */
```

Comment:
```suggestion
#endif /* MBEDTLS_USE_PSA_CRYPTO &&
                PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY &&
                !MBEDTLS_ECP_C */
```

Reply:
Good point. I just removed the comment, as the corresponding `#if` is only 2 lines above, with no nesting, so I don't think the comment was useful. (See [our coding standard](https://mbed-tls.readthedocs.io/en/latest/kb/development/mbedtls-coding-standards/#preprocessor-directives): "Add a comment to the `#endif` directive if the distance to the opening directive is bigger than a few lines or contains other directives".)